crates/bevy_ecs/src/system/combinator.rs

```diff
@@ -83,6 +83,10 @@ use super::{ReadOnlySystem, System};
 /// # assert!(world.resource::<RanFlag>().0);
 /// # world.resource_mut::<RanFlag>().0 = false;
 /// ```
+#[diagnostic::on_unimplemented(
+    message = "`{Self}` can not combine systems `{A}` and `{B}`",
```

Comment:
```suggestion
    message = "`{Self}` can not combine systems `{A}` and `{B}`: the input and output types must match.",
```

Reply:
I agree with the additional information, but I'm going to place it in a `note` instead just to keep the primary message clean:

_"the inputs and outputs of `{A}` and `{B}` are not compatible with this combiner"_